l1-contracts/contracts/state-transition/StateTransitionManager.sol

```diff
@@ -75,8 +84,30 @@ contract StateTransitionManager is IStateTransitionManager, ReentrancyGuard, Own
         _;
     }
 
+    /// @notice Returns all the registered hyperchain addresses
+    function getAllHyperchains() public view override returns (address[] memory chainAddresses) {
+        uint256[] memory keys = hyperchainMap.keys();
+        chainAddresses = new address[](keys.length);
+        for (uint256 i = 0; i < keys.length; i++) {
+            // slither-disable-next-line unused-return
+            (, chainAddresses[i]) = hyperchainMap.at(i);
+        }
+    }
+
+    /// @notice Returns all the registered hyperchain chainIDs
+    function getAllHyperchainChainIDs() public view override returns (uint256[] memory) {
+        return hyperchainMap.keys();
+    }
+
+    /// @notice Returns the address of the hyperchain with the corresponding chainID
+    function hyperchain(uint256 _chainId) public view override returns (address chainAddress) {
```

Comment:
a bit of opinionated nit

```suggestion
    function getHyperchainByChainId(uint256 _chainId) public view override returns (address chainAddress) {
```

Reply:
What about `getHyperchain` as shorter and consistent with BridgeHub version of this func?